test/update_coeffs_testutils.jl

```diff
@@ -0,0 +1,28 @@
+# Utilities for testing update coefficient behaviour with state-dependent (i.e. dependent on u/p/t) functions
+
+mutable struct WrapFunc{F,U,P,T}
+    func::F
+    u::U
+    p::P
+    t::T
+end
+
+(w::WrapFunc)(u) = sum(w.u) * w.p * w.t * w.func(u) 
+function (w::WrapFunc)(v, u) 
```

Comment:
@vpuri3 can you take a look at how this operator is using `u` as part of its state, and applying it to a different RHS `v`? Is this something that is invalid in your opinion? Because your recent changes don't work with this -- this also relates to the discussion in https://github.com/SciML/SciMLOperators.jl/issues/57.

Reply:
(This is a separate, SciMLOperators issue, so resolved for our purposes here)